docs/porting/security/psa_targets_in_mbed-os.md

```diff
@@ -0,0 +1,561 @@
+and [porting targets](../porting/porting-targets.html)):
+
+* `inherits`: PSA generic target `PSA_V8_M`, unless the target has to inherit from
+  one of its family targets.
+* `tfm_target_name`: Target name in TF-M.
+* `tfm_bootloader_supported`: If TF-M bootloader is supported by the target.
+
+    The supported values are `true` and `false`.
+* `tfm_supported_toolchains`: Supported TF-M toolchains.
+
+    The supported values are `ARMCLANG` and `GNUARM`.
+* `tfm_default_toolchain`: Default TF-M toolchain.
+
+     The supported values are `ARMCLANG` and `GNUARM`.
+
+* `tfm_delivery_dir`: The directory to which TF-M binary will be copied.
+* `TFM_OUTPUT_EXT`: Optional attribute that indicates the output extension of the TF-M secure binary.
+
+The following example shows a PSA-enabled Armv8-M PSA target, `ARM_MUSCA_A1`:
+
+```json
+    "ARM_MUSCA_A1": {
+        "inherits": [
+            "PSA_V8_M"
+        ],
+        "default_toolchain": "ARMC6",
+        "forced_reset_timeout": 7,
+        "release_versions": [
+            "5"
+        ],
+        "core": "Cortex-M33-NS",
+        "supported_toolchains": [
+            "ARMC6",
+            "GCC_ARM",
+            "IAR"
+        ],
+        "device_has_add": [
+            "INTERRUPTIN",
+            "LPTICKER",
+            "SERIAL",
+            "SLEEP",
+            "USTICKER"
+        ],
+        "macros_add": [
+            "__STARTUP_CLEAR_BSS",
+            "MBED_FAULT_HANDLER_DISABLED",
+            "CMSIS_NVIC_VIRTUAL",
+            "LPTICKER_DELAY_TICKS=1",
+            "MBED_MPU_CUSTOM"
+        ],
+        "extra_labels_add": [
+            "ARM_SSG",
+            "MUSCA_A1",
+            "MUSCA_A1_NS"
+        ],
+        "post_binary_hook": {
+            "function": "ArmMuscaA1Code.binary_hook"
+        },
+        "secure_image_filename": "tfm_s.bin",
+        "tfm_target_name": "MUSCA_A",
+        "tfm_bootloader_supported": true,
+        "tfm_default_toolchain": "ARMCLANG",
+        "tfm_supported_toolchains": [
+            "ARMCLANG",
+            "GNUARM"
+        ],
+        "tfm_delivery_dir": "TARGET_ARM_SSG/TARGET_MUSCA_A1"
+    }
+```
+
+Please note the config options `macros_add`, `extra_labels_add` and `device_has_remove`. To add or remove `macros`, `extra_labels` or target capabilities, a PSA target definition must use
+[macros/extra_labels/device_has]`_add` or [macros/extra_labels/device_has]`_remove` (not `macros`, `extra_labels` or `device_has`).
+
+To add or remove a feature, use `feature_`[add/remove].
+
+By default, a TF-M build generates a `bin` file. If the target requires a `hex` file, you need to add the attribute `"TFM_OUTPUT_EXT": "hex"` to the target definition. The build script will convert `bin` to `hex`, then copy the `hex` to `tfm_delivery_dir`. You must also update `secure_image_filename` to match the new file extension.
```

Comment:
```suggestion
By default, a TF-M build generates a `bin` file. If the target requires a `hex` file, you need to add the attribute `"tfm_output_ext": "hex"` to the target definition. The build script will convert `bin` to `hex`, then copy the `hex` to `tfm_delivery_dir`. You must also update `secure_image_filename` to match the new file extension.
```

Reply:
Unfortunately, we can't change it here. The code contains capitalised text.